src/Uns.ts

```diff
@@ -260,25 +260,24 @@ export default class Uns extends NamingService {
     return !(await this.isRegistered(domain));
   }
 
-  async registryAddress(domain: string): Promise<string> {
+  async registryAddress(domainOrNamehash: string): Promise<string> {
     let namehash: string;
-    if (domain.startsWith('0x')) {
-      namehash = this.namehash(domain);
-    } else {
-      if (!this.checkDomain(domain)) {
-        throw new ResolutionError(ResolutionErrorCode.UnsupportedDomain, {
-          domain,
-        });
-      }
-
-      const tld = domain.split('.').pop();
-      namehash = this.namehash(tld!);
+    if (domainOrNamehash.startsWith('0x')) {
+      namehash = this.namehash(domainOrNamehash);
+    } else if (!this.checkDomain(domainOrNamehash)) {
+      throw new ResolutionError(ResolutionErrorCode.UnsupportedDomain, {
+        domain: domainOrNamehash,
+      });
     }
+
+    const tld = domainOrNamehash.split('.').pop();
+    namehash = this.namehash(tld!);
```

Comment:
I would pass domain hash instead of just TLD hash to `registryOf`: it is possible that domains from same TLD will live on different registries in the future. Lets be resistant to that.

```suggestion
```